config/database.rb

```diff
@@ -0,0 +1,15 @@
+# set the database based on the current environment
+database_name = "chitter-#{Chitter.environment}"
+db = URI.parse( ENV['DATABASE_URL'] || "postgres://localhost/#{database_name}")
+
+
+# connect ActiveRecord with the current database
+ActiveRecord::Base.establish_connection(
+ :adapter => db.scheme == "postgres" ? "postgresql" : db.scheme,
+ :host => db.host,
+ :port => db.port,
```

Comment:
Never seen this before but it looks interesting, keen to understand how it works.